inception-search-core/src/main/java/de/tudarmstadt/ukp/inception/search/SearchResult.java

```diff
@@ -126,4 +126,65 @@ public void setDocumentTitle(String aDocumentTitle)
     {
         documentTitle = aDocumentTitle;
     }
+
+    @Override
+    public int hashCode()
+    {
+        final int prime = 31;
+        int result = 1;
+        result = prime * result + ((documentTitle == null) ? 0 : documentTitle.hashCode());
+        result = prime * result + ((leftContext == null) ? 0 : leftContext.hashCode());
+        result = prime * result + offsetEnd;
+        result = prime * result + offsetStart;
+        result = prime * result + ((rightContext == null) ? 0 : rightContext.hashCode());
+        result = prime * result + ((text == null) ? 0 : text.hashCode());
+        result = prime * result + tokenLength;
+        result = prime * result + tokenStart;
+        return result;
```

Comment:
I should be enough to compare the begin/end character offsets and the document identity. Other informations such as left/right context, text, length, etc. is depending on that.

Same in the equals method.

Although I do not have a particularly strong opinion on that over leaving it is as is.

Reply:
The contexts could be different, and nothing stops one erroneous method of putting incorrect information in the other fields in a such a way that the dependency that you refer to is broken. In that case, all the fields would be important to technically differentiate between two instances. I would let it as is.

Reply:
I would discern between authoritative information and auxiliary information. I would count the offsets and IDs as authoritative and the text/context strings as auxiliary - reason being that the text/context can be reconstructed from the CAS when IDs/offsets are known. The hash/equals should IMHO rely only on authoritative information.